### DISCUSSION TITLE
Can I use my own custom installer ? 🪛

### DISCUSSION THREAD

**Discussion by akhildevelops** [python-poetry/poetry#7331] [General] [answered]
Poetry plugin is a wonderful feature for extending functionality.

I understand that currently poetry object is exposed.

Can I also tweak Installer object or use a custom installer through the plugin feature ?

Poetry's Installer: https://github.com/python-poetry/poetry/blob/3fe8838659415f7495d415503ff64eea1aa0bf57/src/poetry/installation/installer.py#L35

I'm thinking of using my own custom based solver.

Poetry's Solver:
https://github.com/python-poetry/poetry/blob/3fe8838659415f7495d415503ff64eea1aa0bf57/src/poetry/puzzle/solver.py#L38

**Comment by radoering** [comment, marked as answer]
That's probably not the intended use of plugins and doubt whether that's a good idea but you can patch almost every part of poetry in a plugin. However, you should be aware that updating to a new version of poetry becomes harder the more you are relying on specific internals. See https://python-poetry.org/docs/master/plugins/#maintaining-a-plugin

**Comment by akhildevelops** [comment]
Thanks @radoering
